### ISSUE TITLE
Internal DNS resolving .svc to public .svc domain instead of .svc.cluster.local when member of RH IDM Domain

### ISSUE THREAD

**Issue by mlbiam** [openshift/origin#17459]
[provide a description of the issue]
Running openshift-enterprise installed with ansible (generated from hosts from quick install) on RHEL 7.4 thats joined to RHIDM domain (RHIDM domain is NOT cross forrest trusted with an AD domain but is forwarding DNS requests).  When I attempt a build, the build completes but fails to push into the internal registry with the error `https://docker-registry.default.svc:5000/v1/ ping: dial tcp 92.242.140.21:5000: i/o timeout`.  Doing a quick DNS check shows that  `docker-registry.default.svc.` resolves to `92.242.140.21` BUT `docker-registry.default.svc.cluster.local' resolves to `172.30.18.163` which is the IP of the registry.  i found #16097 but the fixes don't fix my resolution issue.

##### Version

[root@osmaster2 ~]# oc version
oc v3.6.173.0.49
kubernetes v1.6.1+5115d708d7
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://console.openshift.lan:8443
openshift v3.6.173.0.49
kubernetes v1.6.1+5115d708d7
[root@osmaster2 ~]# openshift version

##### Steps To Reproduce
1.  Install RHIDM on RHEL 7.4 (domain name is  rhelent.lan)
2.  Setup 3 servers on RHEL 7.4, join to RHIDM domain
3.  Install openshift enterprise 3.6 with 1 master, 2 nodes
4.  Run an s2i build (I am using https://github.com/TremoloSecurity/openunison-qs-openshift)

##### Current Result

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:31 min
[INFO] Finished at: 2017-11-26T18:51:46+00:00
[INFO] Final Memory: 49M/199M
[INFO] ------------------------------------------------------------------------
Deleting build artifacts
Copying war file...
Creating /usr/local/openunison/work
Removing source...
Removing maven repo...
Removing maven...
Done.
Pushing image docker-registry.default.svc:5000/openunison/openunison:latest ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Warning: Push failed, retrying in 5s ...
Registry server Address: 
Registry server User Name: serviceaccount
Registry server Email: serviceaccount@example.org
Registry server Password: <<non-empty>>
error: build error: Failed to push image: After retrying 6 times, Push image still failed due to error: Get https://docker-registry.default.svc:5000/v1/_ping: dial tcp 92.242.140.21:5000: i/o timeout

##### Expected Result

image from s2i pushed to internal registry

##### Additional Information
[root@osmaster2 ~]# dig docker-registry.default.svc

; <<>> DiG 9.9.4-RedHat-9.9.4-51.el7 <<>> docker-registry.default.svc
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 4488
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;docker-registry.default.svc.	IN	A

;; ANSWER SECTION:
docker-registry.default.svc. 0	IN	A	92.242.140.21

;; Query time: 31 msec
;; SERVER: 192.168.3.215#53(192.168.3.215)
;; WHEN: Sun Nov 26 14:46:55 EST 2017
;; MSG SIZE  rcvd: 72

[root@osmaster2 ~]# dig docker-registry.default.svc.cluster.local 

; <<>> DiG 9.9.4-RedHat-9.9.4-51.el7 <<>> docker-registry.default.svc.cluster.local
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 22877
;; flags: qr aa rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;docker-registry.default.svc.cluster.local. IN A

;; ANSWER SECTION:
docker-registry.default.svc.cluster.local. 30 IN A 172.30.18.163

;; Query time: 1 msec
;; SERVER: 192.168.3.215#53(192.168.3.215)
;; WHEN: Sun Nov 26 14:47:01 EST 2017
;; MSG SIZE  rcvd: 75

[root@osmaster2 ~]# cat /etc/resolv.conf 
# nameserver updated by /etc/NetworkManager/dispatcher.d/99-origin-dns.sh
# Generated by NetworkManager
#search rhelent.lan
domain cluster.local
search default.svc.cluster.local cluster.local
nameserver 192.168.3.215

**Comment by mlbiam** [comment]
so after a fun twitter debugging session trying various combinations we came to realize that the pod is correctly resolving `docker-registry.default.svc` by first adding `cluster.local` to make `docker-registry.default.svc.cluster.local` BUT docker push uses the docker daemon on the host so its using the host's `/etc/resolv.conf`.  No combination of `search` or `domain` configurations made it work properly.  I also rebuilt without RHIDM (which turned out to be a red herring), tried removing the domain name from the configured host name, which also didn't working.  The only solution that worked was to hard code the host name into `/etc/hosts` on all my nodes (which seems like the absolute worst way to handle this issue but its working)
